### PR TITLE
New emulator methods

### PR DESCRIPTION
**Pull request by vrunoa**
Add the new methods we have for emulators

### CODE REVIEW

**Comment by vrunoa** [comment]
@jlipps yeah definitely. Wanna drive this ? I would like to see how you guys handle this in the past.

**Comment by vrunoa** [comment]
* _powerAC, powerCapacity_ would be used for all the this other platforms appium supports: ios, windows desktop, osx desktop. Not sure if this would make sense for youi.tv
* same for _fingerprint_, mobile platforms + some windows and osx machines has fingerprint integrated
* I believe _sendSMS, gsmSignal, gsmCall, gsmVoice_ would be only available for mobile platforms

**Comment by jlipps** [comment]
Yes, we can drive this, however I think it'd be fun for you to have a look at the current [mobile spec](https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md) to get a sense of how things are written. Care to take an initial stab and then we can walk through the rest of the process together?

**Comment by vrunoa** [comment]
Sure @jlipps i'll take a look tonight. Ping me whenever you want(and have the time) after that

File: lib/commands/actions.js [hunk 1]
```diff
@@ -182,6 +182,41 @@ commands.sendSMS = async function (phoneNumber, message) {
   await this.adb.sendSMS(phoneNumber, message);
 };
 
+commands.gsmCall = async function (phoneNumber, action) {
```
Comment (mykola-mokhnach) [inline lib/commands/actions.js]: should these command return something?

Comment (vrunoa) [inline lib/commands/actions.js]: *adb* doesnt return any value when calling it, so I guess not. If adbExec fails it will throw an error, right ?

Comment (mykola-mokhnach) [inline lib/commands/actions.js]: Right. It was just a small reminder to make sure we didn't forget anything

**Comment by vrunoa** [comment]
@jlipps would the new methods go under _Other devices features_ section ? https://github.com/SeleniumHQ/mobile-spec/compare/master...vrunoa:master

**Comment by jlipps** [comment]
In this case I'm proposing that they each get their own sections in the spec, not that they be left out of the spec :-)

**Comment by vrunoa** [comment]
Something like this? but well written 😄 
https://github.com/SeleniumHQ/mobile-spec/compare/master...vrunoa:master#diff-ee2352b7c156f97e44f211c294a46f2dR278

**Comment by jlipps** [comment]
@vrunoa that looks great. The only thing missing is the actual endpoints and request/response values you would suggest.

**Comment by vrunoa** [comment]
@jlipps this is ready. I'll add the endpoints on the appium-base-driver

**Comment by filmaj** [comment]
Woo! This is rad. Thanks for taking this on.

**Comment by vrunoa** [comment]
@jlipps going to include the new network speed method also on this PR. So, it's not ready to merge.

File: lib/android-helpers.js [hunk 2]
```diff
@@ -59,9 +59,25 @@ helpers.prepareEmulator = async function (adb, opts) {
       return;
     }
   }
+  /* TODO - update appium-adb version to latest once https://github.com/appium/appium-adb/pull/245
+  is merged adn published, then finish the network speed implementation
+  if (_.isUndefined(opts.networkSpeed)) {
+    let networkSpeed = this.ensureNetworkSpeed(adb, opts.networkSpeed);
+    avdArgs = _.concat(avdArgs, ['-netspeed', networkSpeed]);
```
Comment (jlipps) [inline lib/android-helpers.js]: should now be able to do:

```js
avdArgs = [...avdArgs, '-netspeed', networkSpeed];
```

**Comment by vrunoa** [comment]
@imurchie @jlipps ready for another review.

**Comment by imurchie** [comment]
Rebase and fix conflicts, and we should be good!

**Pull request merged**